crates/libs/wallet/src/types/compact_wad.rs

```diff
@@ -28,6 +36,34 @@ pub enum Error {
     InvalidBase64(#[from] bitcoin::base64::DecodeError),
     #[error("failed to deserialize the CBOR wad representation: {0}")]
     InvalidCbor(#[from] ciborium::de::Error<std::io::Error>),
+    #[error("invalid dleq proof")]
+    DleqProofError,
+    #[error("Custom Error: {0}")]
+    Custom(String),
+
+    /// Tonic gRPC status error
+    #[error("gRPC Error: {0}")]
+    Grpc(#[from] Status),
+
+    /// NUTs error
+    #[error("Nuts Error: {0}")]
+    Nuts(#[from] nuts::Error), // A general nuts error
+
+    /// Specific NUT02 Keysets error
+    #[error("NUT02 Keysets Error: {0}")]
+    Nut02(#[from] nuts::nut02::Error),
+
+    /// Specific NUT01 error
+    #[error("NUT01 Keys Error: {0}")]
+    Nut01(#[from] nuts::nut01::Error),
+}
+
+#[derive(Debug, Clone)]
+pub struct DleqVerificationResult {
+    pub total_proofs: usize,
+    pub valid_proofs: usize,
+    pub invalid_proofs: Vec<DleqProofError>,
+    pub is_fully_valid: bool,
```

Comment:
`is_fully_valid` is just `invalid_proofs.is_empty()`
`valid_proofs` is `total_proofs - invalid_proofs.len()`

Implement those are getters rather than adding fields to the struct